src/components/announcer/Announcer.js

```diff
@@ -6,157 +6,110 @@
     return (
-        <SnackbarContent
-            className={classNames(classes[variant], classes.margin)}
-            aria-describedby="client-snackbar"
-            message={
-                <span id="client-snackbar" className={classes.message}>
-                    <Icon
-                        className={classNames(
-                            classes.icon,
-                            classes.iconVariant
-                        )}
-                    />
-                    {message}
-                </span>
-            }
-            action={[
-                customAction ? customAction() : null,
-                <IconButton
-                    key="close"
-                    aria-label="Close"
-                    color="inherit"
-                    className={classes.close}
-                    onClick={() => {
-                        console.log("close");
-                        onClose();
-                    }}
-                >
-                    <CloseIcon className={classes.icon} />
-                </IconButton>,
-            ]}
-        />
+        <Alert
+            elevation={6}
+            variant="filled"
+            severity={variant}
+            onClose={onClose}
+            action={action}
+            style={{ color: getTextColor(theme, variant) }}
+        >
+            {message}
+        </Alert>
     );
 }
 
 MySnackbarContent.propTypes = {
-    classes: PropTypes.object.isRequired,
-    className: PropTypes.string,
     message: PropTypes.node,
     onClose: PropTypes.func,
     variant: PropTypes.oneOf(["success", "warning", "error", "info"])
         .isRequired,
+    action: PropTypes.node,
 };
 
-const MySnackbarContentWrapper = withStyles(styles1)(MySnackbarContent);
-
-const styles2 = (theme) => ({});
-
-class Announcer extends React.Component {
-    constructor(props) {
-        super(props);
-    }
-
-    render() {
-        const {
-            classes,
-            message,
-            variant,
-            duration,
-            horizontal,
-            vertical,
-            onClose,
-            open,
-            customAction,
-        } = this.props;
-        if (message) {
-            return (
-                <div>
-                    <Snackbar
-                        key={message}
-                        anchorOrigin={{
-                            vertical: vertical ? vertical : "bottom",
-                            horizontal: horizontal ? horizontal : "left",
-                        }}
-                        open={open}
-                        autoHideDuration={duration ? duration : 6000}
+function Announcer(props) {
+    const {
+        message,
+        variant,
+        duration,
+        horizontal,
+        vertical,
+        onClose,
+        open,
+        CustomAction,
+    } = props;
+    const theme = useTheme();
+    if (message) {
+        return (
+            <div>
+                <Snackbar
+                    key={message}
```

Comment:
I don't think this `key` prop is necessary? This might be why this component would only accept a string message instead of an object returned from `getMessage`.  I might be missing something though.

Reply:
They `key` prop shouldn't be needed unless this component is in an array.
This `Snackbar` isn't directly inside an array since it's part of the `Announcer` return.

I thought the `key` wouldn't have any effect outside of the array requirement, though.
I think the issue with a string vs. an object `message` is only due to the way it's ultimately called inside the `Alert`.

Reply:
`Key` is needed. Please see - https://material-ui.com/api/snackbar/ 

`When displaying multiple consecutive Snackbars from a parent rendering a single <Snackbar/>, add the key prop to ensure independent treatment of each message. e.g. <Snackbar key={message} />, otherwise, the message may update-in-place and features such as autoHideDuration may be canceled.`

In line with, removing `key` broke this component in SB.